src/main/scala/com/amazon/deequ/analyzers/MaxLength.scala

```diff
@@ -41,4 +41,6 @@ case class MaxLength(column: String, where: Option[String] = None)
   }
 
   override def filterCondition: Option[String] = where
+
+  private def criterion: Column = length(conditionalSelection(column, where)).cast(DoubleType)
```

Comment:
This is a private method here and a public method in Completeness. If it is required in each analyzer, should it be part of a base class?

Reply:
the other one is used by a test that constructs an expected object and compares the result of Completeness to that object. It shouldn't be public because I don't think anything in the code should invoke it. I'll make it package-private inside deequ and mark it as `@VisibleForTesting` to point out it's an internal detail for the analyzer